### PR TITLE
🐛Git read history from fallback repo with multiple docset

### PR DESCRIPTION
**Pull request by live1206**
https://dev.azure.com/ceapex/Engineering/_workitems/edit/134116

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5237)

### CODE REVIEW

File: src/docfx/build/context/Input.cs [hunk 1]
```diff
@@ -188,23 +190,31 @@ FilePath CreateFilePath(string path)
...
         }
 
-        private (string basePath, string path, string commit) ResolveFilePath(FilePath file)
+        private (string repoPath, string pathToRepo, string commit) ResolveFilePath(FilePath file)
         {
             switch (file.Origin)
             {
                 case FileOrigin.Default:
-                    return (_docsetPath, file.Path, file.Commit);
+                    return (_repoPath, PathUtility.NormalizeFile(Path.GetRelativePath(_repoPath, PathUtility.NormalizeFile(Path.Combine(_docsetPath, file.Path)))), file.Commit);
```
Comment (yufeih) [inline src/docfx/build/context/Input.cs]: this can be a folder without git, see #5238

File: src/docfx/docset/RepositoryProvider.cs [hunk 2]
```diff
@@ -3,7 +3,9 @@
 
 using System;
 using System.Collections.Concurrent;
+using System.ComponentModel.DataAnnotations;
```
Comment (yufeih) [inline src/docfx/docset/RepositoryProvider.cs]:
```suggestion
~~using System.ComponentModel.DataAnnotations;~~
```

File: src/docfx/build/Build.cs [hunk 3]
```diff
@@ -40,6 +40,7 @@ private static async Task BuildDocset(string docsetPath, string outputPath, Comm
                     var repository = repositoryProvider.GetRepository(FileOrigin.Default);
                     Telemetry.SetRepository(repository?.Remote, repository?.Branch);
                     var locale = LocalizationUtility.GetLocale(repository, options);
+                    var docsetSourceFolder = repository is null ? string.Empty : PathUtility.NormalizeFolder(Path.GetRelativePath(repository.Path, docsetPath));
```
Comment (mingwli) [inline src/docfx/build/Build.cs]: > var docsetSourceFolder =  [](start = 19, length = 26)

where is this used?

Comment (live1206) [reply src/docfx/build/Build.cs]: can be removed

**Pull request merged**